broker-cli/commands/wallet.js

```diff
@@ -130,29 +130,42 @@ async function commitBalance (args, opts, logger) {
     const { uncommittedBalance } = balances.find(({ symbol: s }) => s === symbol)
 
     const totalUncommittedBalance = Big(uncommittedBalance)
+
     if (totalUncommittedBalance.eq(0)) {
       return logger.info('Your current uncommitted balance is 0, please add funds to your daemon')
     }
-    // We try to take the `Math.min` total here between 2 Big numbers due to a
+
+    // We try to take the lowest total here between 2 Big numbers due to a
     // commit limit specified as MAX_CHANNEL_BALANCE
     let maxSupportedBalance = uncommittedBalance
 
     if (totalUncommittedBalance.gt(ENUMS.MAX_CHANNEL_BALANCE)) {
       maxSupportedBalance = ENUMS.MAX_CHANNEL_BALANCE
     }
 
+    // We use this for normalization of the amount from satoshis to a decimal
+    // value
     const divideBy = currencyConfig.find(({ symbol: configSymbol }) => configSymbol === symbol).quantumsPerCommon
 
+    // Amount is specified in currency instead of satoshis
+    if (amount) {
+      maxSupportedBalance = (amount * divideBy)
+    }
+
     logger.info(`For your knowledge, the Maximum supported balance at this time is: ${Big(ENUMS.MAX_CHANNEL_BALANCE).div(divideBy)} ${symbol}`)
     logger.info(`Your current uncommitted wallet balance is: ${Big(uncommittedBalance).div(divideBy)} ${symbol}`)
 
-    const answer = await askQuestion(`Are you OK committing ${Big(maxSupportedBalance).div(divideBy)} ${symbol} to SparkSwap? (Y/N)`)
+    const answer = await askQuestion(`Are you OK committing ${Big(maxSupportedBalance).div(divideBy)} ${symbol} to sparkswap? (Y/N)`)
 
     if (!ACCEPTED_ANSWERS.includes(answer.toLowerCase())) return
 
+    if (Big(maxSupportedBalance).gte(uncommittedBalance)) {
+      throw new Error(`Amount specified is larger than your current uncommitted balance of ${Big(uncommittedBalance).div(divideBy)} ${symbol}`)
+    }
+
     await client.walletService.commitBalance({ balance: maxSupportedBalance.toString(), symbol })
 
-    logger.info('Successfully committed balance to SparkSwap Relayer!')
+    logger.info('Successfully committed balance to sparkswap Relayer!')
```

Comment:
this is definitely not for this ticket, but I think we need some way to check on the status of a commitment to the Relayer. Particularly since we've seen most of the problems crop up when one out of the two channels are open - that's a confusing state for a user unless they know A LOT about the inner workings.

Reply:
(I opened a ticket: https://trello.com/c/W4XTd5wl/555-better-user-feedback-on-balance-commitment-status)